crates/store/blobs/src/lib.rs

```diff
@@ -74,8 +75,15 @@ impl BlobManager {
 
         let blobs = try_stream!({
             let mut buf = {
-                let mut buf = Vec::with_capacity(CHUNK_SIZE);
+                let mut buf: Vec<MaybeUninit<u8>> = Vec::with_capacity(CHUNK_SIZE);
                 unsafe { buf.set_len(CHUNK_SIZE) };
+
+                let buf = unsafe {
+                    // Convert the Vec<MaybeUninit<u8>> into a Vec<u8> once it's
+                    // fully-initialised
+                    Vec::from_raw_parts(buf.as_mut_ptr().cast::<u8>(), buf.len(), buf.capacity())
+                };
```

Comment:
You just created two `Vec`s backed by the same memory, that'll double-free..

The current behavior was intentional, as this is a byte array, zero-filled or having arbitrary data, it doesn't matter, it's still a valid byte array. More complex types could be invalidly represented, but not this.

The logic that follows determines the legality/validity of slices of the buffer as we only read slices we previously wrote. It has nothing to do with it's initialization.

But to correct the patch above, either of these would've been correct:

```suggestion
let mut buf = vec![0; CHUNK_SIZE];
```

or, reverting back to what we had before, and finally zero-filling it after allocation:

```suggestion
let mut buf = Vec::with_capacity(CHUNK_SIZE);
unsafe { buf.set_len(CHUNK_SIZE) };
std::ptr::write_bytes(buf.as_mut_ptr(), 0, CHUNK_SIZE)
```

Reply:
Basically what I already said in our chat on the subject 🙂 You are correct - I had initially simply initialised it, and then thought that by using `MaybeUninit` it would be a more elegant approach, and more performant. But I've gone back to the first fix, as it's just simpler and removes the unsafe code entirely.